android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/MatchingEngineHelper.java

```diff
@@ -80,8 +88,203 @@ public MatchingEngineHelper(Context context, View view) {
+                break;
+            case HEALTH_CHECK_OK:
+                Log.i(TAG, "AppInst Health is OK");
+                break;
+            case UNRECOGNIZED:
+                // fall through
+            default:
+                Log.i(TAG, "AppInst Health event: " + event.getHealthCheck());
+        }
+    }
+
+    void handleCloudletMaintenance(AppClient.ServerEdgeEvent event) {
+        if (event.getEventType() != AppClient.ServerEdgeEvent.ServerEventType.EVENT_CLOUDLET_MAINTENANCE) {
+            return;
+        }
+
+        mMatchingEngineResultsInterface.showMessage("Received: " + event.getEventType());
+
+        switch (event.getMaintenanceState()) {
+            case NORMAL_OPERATION:
+                Log.i(TAG, "Maintenance state is all good!");
+                break;
+            default:
+                Log.i(TAG, "Server maintenance: " + event.getMaintenanceState());
+        }
+    }
+
+    void handleCloudletState(AppClient.ServerEdgeEvent event) {
+        if (event.getEventType() != AppClient.ServerEdgeEvent.ServerEventType.EVENT_CLOUDLET_STATE) {
+            return;
+        }
+
+        mMatchingEngineResultsInterface.showMessage("Received: " + event.getEventType());
+
+        switch (event.getCloudletState()) {
+            case CLOUDLET_STATE_INIT:
+                Log.i(TAG, "Cloudlet is not ready yet. Wait or FindCloudlet again.");
+                break;
+            case CLOUDLET_STATE_NOT_PRESENT:
+                Log.i(TAG, "Cloudlet is not present.");
+            case CLOUDLET_STATE_UPGRADE:
+                Log.i(TAG, "Cloudlet is upgrading");
+            case CLOUDLET_STATE_OFFLINE:
+                Log.i(TAG, "Cloudlet is offline");
+            case CLOUDLET_STATE_ERRORS:
+                Log.i(TAG, "Cloudlet is offline");
+            case CLOUDLET_STATE_READY:
+                // Timer Retry or just retry.
+                doEnhancedLocationUpdateInBackground(mMatchingEngineResultsInterface.getLocationForMatching());
+                break;
+            case CLOUDLET_STATE_NEED_SYNC:
+                Log.i(TAG, "Cloudlet data needs to sync.");
+                break;
+            default:
+                Log.i(TAG, "Not handled");
+        }
+    }
+    // Only the app knows with any certainty which AppPort (and internal port array)
+    // it wants to test, so this is in the application.
+    // TODO: This has things hard-coded to allow a demo even though the app can't reach
+    //  the app instances. When the DME code is released and doesn't have to be run on EdgeBox,
+    //  this needs to be reworked.
+    void handleLatencyRequest(AppClient.ServerEdgeEvent event) {
+        if (event.getEventType() != AppClient.ServerEdgeEvent.ServerEventType.EVENT_LATENCY_REQUEST) {
+            return;
+        }
+
+        mMatchingEngineResultsInterface.showMessage("Received: " + event.getEventType());
+
+        AsyncTask.execute(new Runnable() {

```

Comment:
We should probably document this for app developers, so they can just copy and paste this code (or the code when it is reworked when DME code is released). Or @lgarner can create a single function in the SDK that applications can just call: HandleLatencyRequest(cloudlet, portNum, testType) and that will also call postLatencyResult for the application.